pkg/extraconfig/extraconfig.go

```diff
@@ -77,8 +77,9 @@ func GetExtraConfig() (*ExtraConfig, error) {
 			extraConfig.AwsProfile = val
 		case "awsCredentialFile":
 			extraConfig.AwsCredentialFile = val
+		default:
+			os.Setenv(key, val)
```

Comment:
Any reason/use case we want to allow this behaviour?

Reply:
we want to allow customer to config env vars in `extracfg.txt` and  read by AOC. see https://github.com/aws-observability/aws-otel-collector/issues/451. I think the customer doesn't have a good way to config env when running AOC RPM/Deb/MSI.